src/client/Client.cc

```diff
@@ -15360,18 +16128,24 @@ int Client::_rename(Inode *fromdir, const char *fromname, Inode *todir, const ch
   InodeRef target;
   MetaRequest *req = new MetaRequest(op);
 
-  filepath from;
-  fromdir->make_nosnap_relative_path(from);
-  from.push_dentry(fromname);
   filepath to;
-  todir->make_nosnap_relative_path(to);
-  to.push_dentry(toname);
-  req->set_filepath(to);
+  Dentry *de;
+  int r = _prepare_req_path(todir, req, to, toname, true, &de);
+  if (r < 0) {
+    delete req;
+    return r;
+  }
+
+  filepath from;
+  Dentry *oldde;
+  r = _prepare_req_path(fromdir, req, from, fromname, false, &oldde);
+  if (r < 0) {
+    delete req;
+    return r;
+  }
   req->set_filepath2(from);
-  req->set_alternate_name(std::move(alternate_name));
 
-  Dentry *oldde = get_or_create(fromdir, fromname);
-  Dentry *de = get_or_create(todir, toname);
+  req->set_alternate_name(std::move(alternate_name));
```

Comment:
Was this missing

```suggestion
  req->set_alternate_name(std::move(alternate_name));
  todir->gen_inherited_fscrypt_auth(&req->fscrypt_auth);
```

? I've fixed this in my branch.

Reply:
Yep missed here too